crates/router/src/connector/coinbase/transformers.rs

```diff
@@ -250,6 +252,17 @@ pub struct CoinbaseConnectorMeta {
     pub pricing_type: String,
 }
 
+impl TryFrom<&Option<pii::SecretSerdeValue>> for CoinbaseConnectorMeta {
```

Comment:
This changes will impact previous merchants who are using coinbase, This PR needs data migrations. Could you please refer https://github.com/juspay/hyperswitch/pull/2746 description for more info.

Reply:
[Issue#2899](https://github.com/juspay/hyperswitch/issues/2899) does not mention any data migrations.
I've checked that #2746, and in that case, the migration involved replacing `terminalId` with `terminal_id` due to the removal of `#[serde(rename_all = "camelCase")]` [here](https://github.com/juspay/hyperswitch/pull/2746/files#diff-bd16bd6d296808eacd72cdeb8a8bfbe54bc788b30b844301e136c4396a53eeceL421-R425).

However, my issue is different. If I understand correctly, this issue is about empty JSON for `CoinbaseConnectorMeta` or the absence of the `pricing_type` attribute in the JSON, and the goal is to add validations to prevent storing invalid `CoinbaseConnectorMeta`.

So, I am not sure about the migration here. If empty JSON or JSON without `pricing_type` is stored, how should the migration to the correct `CoinbaseConnectorMeta` be handled? What value should be set for `pricing_type`?

Reply:
No worries, We will take this up as these are data migrations. FYI, I was talking about existing merchants, we need to add `pricing_type` in metadata and set it to empty string. In that case, when merchant tries to make payment connector throws error and they are expected to update the metadata.

Reply:
@srujanchikke aha, ok, thank you, make senses now, so I prepared migration script and simple test demonstration that it works, please see PR description.